Code/CSharpAdvanced/FlowerExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CSharpAdvanced
+{
+    public static class FlowerExtensions
+    {       
+        public static string GenerateRandomColor(this Flower flower)
+        {
+            Random random = new Random();
+            string[] colors = { "Red", "Blue", "Yellow", "Pink", "Purple" };
```

Comment:
I see duplication of colors list :) Please make it const for all class :)

Reply:
done